### PR TITLE
Simplify JDBC span names

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Adding the `<db.operation> <db.name>.<db.sql.table>` idea from https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1409

I think it covers most of the common scenarios (and it's not that complicated).

Spec PR: https://github.com/open-telemetry/opentelemetry-specification/pull/1219

### CODE REVIEW

**Review by trask — COMMENTED** [review]
👍 for moving this POC forward

**Review by iNikem — APPROVED** [review]

**Comment by iNikem** [comment]
I would like to get approval either from @trask or @johnbley

File: javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/normalizer/SqlStatementInfoExtractorTest.groovy [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    'update /*table'                                                  | new SqlStatementInfo(sql, 'UPDATE', null)
+    // Merge
+    'merge into table'                                                | new SqlStatementInfo(sql, 'MERGE', 'table')
+    'merge table (into is optional in some dbs)'                      | new SqlStatementInfo(sql, 'MERGE', 'table')
+    'merge (into )))'                                                 | new SqlStatementInfo(sql, 'MERGE', null)
+    // Unknown operation
+    'and now for something completely different'                      | new SqlStatementInfo(sql, null, null)
+    ''                                                                | new SqlStatementInfo(sql, null, null)
+    null                                                              | new SqlStatementInfo(sql, null, null)
+  }
```
Comment (johnbley) [inline javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/normalizer/SqlStatementInfoExtractorTest.groovy:59]: Maybe add a comment one just to document the known edge case (unless I'm misreading the code and this actually works)?
```
'/*comment*/ select * from table1' -> null/null
```

**Review by johnbley — APPROVED** [review]

File: javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/normalizer/SqlStatementInfoExtractorTest.groovy (hunk 1 above)
Comment (mateuszrzeszutek) [reply javaagent-api/src/test/groovy/io/opentelemetry/javaagent/instrumentation/api/db/normalizer/SqlStatementInfoExtractorTest.groovy:59]: This actually works, but I'll add a test case for that anyway.

File: instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java [hunk 2]
```diff
@@ -69,19 +71,44 @@ public Span startSpan(PreparedStatement statement) {
...
-  protected String normalizeQuery(String query) {
-    return JDBCUtils.normalizeSql(query);
+  protected String spanName(DBInfo connection, SqlStatementInfo query, String normalizedQuery) {
+    String dbName = dbName(connection);
+    if (query.getOperation() == null && query.getTable() == null) {
+      return dbName == null ? DB_QUERY : dbName;
+    }
+
+    StringBuilder name = new StringBuilder();
+    name.append(query.getOperation()).append(' ');
```
Comment (trask) [inline instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java:101]: can `query.getOperation()` be `null` here?

maybe a unit test for this method to test the different combinations?

**Review by trask — APPROVED** [review]

File: instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java:101]: It can't - the only possible options are `null, null`, `operation, null`, `operation, table`.

I'll add more unit tests anyway.

File: instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java [hunk 3]
```diff
@@ -69,19 +71,44 @@ public Span startSpan(PreparedStatement statement) {
...
+  protected String normalizeQuery(SqlStatementInfo query) {
+    return query.getFullStatement();
   }
 
   @Override
-  protected String normalizeQuery(String query) {
-    return JDBCUtils.normalizeSql(query);
+  protected String spanName(DBInfo connection, SqlStatementInfo query, String normalizedQuery) {
+    String dbName = dbName(connection);
+    if (query.getOperation() == null && query.getTable() == null) {
```
Comment (trask) [inline instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java:96]: I think this makes more clear that query.getOperation() can't be null below
```suggestion
    if (query.getOperation() == null) {
```

**Comment by trask** [comment]
Nice 🎉
